File: backend/models/ticket.go
// Package models defines the Ticket data structure for flight reservations
// in the MindenAirport system.
package models

import "time"

// Ticket represents a flight reservation and booking in the airport system.
// Contains all information related to a passenger's flight booking including
// seat assignment, travel class, pricing, and booking details.
type Ticket struct {
	ID            string    `json:"id"`                      // Unique identifier for the ticket
	AirportUserID string    `json:"airportUserId"`           // ID of the passenger who booked the ticket
	Flight        string    `json:"flight"`                  // Flight ID for this ticket
	SeatNumber    string    `json:"seatNumber,omitempty"`    // Assigned seat number (e.g., "12A")
	TravelClass   string    `json:"travelClass,omitempty"`   // Travel class (Economy, Business, First)
	Price         float64   `json:"price,omitempty"`         // Ticket price in USD
	BookingDate   time.Time `json:"bookingDate,omitempty"`   // Date and time when ticket was booked
	Status        string    `json:"status,omitempty"`        // Ticket status (CONFIRMED, CANCELLED, CHECKED_IN)
	From          string    `json:"from,omitempty"`          // Origin airport code
	To            string    `json:"to,omitempty"`            // Destination airport code
	Gate          string    `json:"gate,omitempty"`          // Departure gate assignment
	BaggageClaim  string    `json:"baggageClaim,omitempty"`  // Baggage claim area for arrival
	DepartureTime string    `json:"departureTime,omitempty"` // Scheduled departure time
}

File: frontend/src/types/ticket.ts
/**
 * Ticket and Booking Type Definitions
 * 
 * This module contains all type definitions related to ticket management,
 * booking operations, and user travel information within the airport system.
 * Supports comprehensive ticket lifecycle from booking to travel completion.
 */

import type { FlightDisplayInfo } from './flight';

/**
 * Core ticket entity representing a flight booking
 */
export interface Ticket {
  /** Unique ticket identifier */
  id: string;
  /** ID of the user who owns this ticket (matches backend field name) */
  airportUserID: string;
  /** Associated flight ID */
  flight: string;
  /** Assigned seat number (optional) */
  seatNumber?: string;
  /** Travel class (economy, business, first, etc.) */
  travelClass?: string;
  /** Ticket price in appropriate currency (optional) */
  price?: number;
  /** Date when the ticket was booked (optional) */
  bookingDate?: string;
  /** Current status of the ticket */
  status?: 'CONFIRMED' | 'CANCELLED' | 'CHECKED_IN';
  /** Origin airport code (optional) */
  from?: string;
  /** Destination airport code (optional) */
  to?: string;
  /** Assigned gate number (optional) */
  gate?: string;
  /** Baggage claim area (optional) */
  baggageClaim?: string;
  /** Departure time for convenience (optional) */
  departureTime?: string;
}

/**
 * Enhanced ticket with enriched flight information
 * 
 * Extends the base Ticket interface with detailed flight data
 * for improved user experience and comprehensive ticket display.
 */
export interface TicketWithFlight extends Ticket {
  /** Enriched flight information associated with the ticket */
  flightInfo?: FlightDisplayInfo;
}

/**
 * User dashboard data structure containing user and travel information
 */
export interface UserDashboard {
  /** User profile information */
  user: {
    /** User's unique identifier */
    id: string;
    /** User's first name */
    firstName: string;
    /** User's last name */
    lastName: string;
    /** User's email address (optional to match backend User type) */
    email?: string;
  };
  /** All tickets associated with the user */
  tickets: TicketWithFlight[];
  /** Upcoming flights (future departure times) */
  upcomingFlights: TicketWithFlight[];
  /** Past flights (completed travel) */
  pastFlights: TicketWithFlight[];
}
